### PR TITLE
pythonPackages.yeelight: init at 0.4.0

### PR DESCRIPTION
**Pull request by puffnfresh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/home-assistant/component-packages.nix [hunk 1]
```diff
@@ -146,7 +146,7 @@
     "light.tikteck" = ps: with ps; [  ];
     "light.tplink" = ps: with ps; [  ];
     "light.xiaomi_miio" = ps: with ps; [  ];
-    "light.yeelight" = ps: with ps; [  ];
+    "light.yeelight" = ps: with ps; [ yeelight ];
```
Comment (puffnfresh) [inline pkgs/servers/home-assistant/component-packages.nix]: Regenerated after adding "yeelight" to Python packages

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -18304,6 +18304,21 @@ EOF
   h11 = callPackage ../development/python-modules/h11 { };
 
   python-docx = callPackage ../development/python-modules/python-docx { };
+
+  yeelight = buildPythonPackage rec {
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Please put the expression in its own file and use `callPackage`.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -18304,6 +18304,21 @@ EOF
...
+    name = "yeelight-${version}";
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Please specify `pname = "yeelight"` instead. `name` will automatically be set by `buildPythonPackage`.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -18304,6 +18304,21 @@ EOF
...
+    version = "0.4.0";
+    src = pkgs.fetchurl {
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: `fetchPypi`

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -18304,6 +18304,21 @@ EOF
...
+      url = "mirror://pypi/y/yeelight/${name}.tar.gz";
+      sha256 = "0v5glz4q52w3p51c2yc3pfjlscbnbjq7lkb6j8cz7kkd5pprpp6f";
+    };
+    propagatedBuildInputs = with self; [ future enum-compat ];
+    meta = {
+      homepage = https://gitlab.com/stavros/python-yeelight;
+      description = "A Python library for controlling YeeLight WiFi RGB bulbs";
+      license = licenses.bsd2;
+    };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Do you want to maintain this package?

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
Nothing happening so closing.

**Pull request closed without merging**